### PR TITLE
Fix vmem issue #968

### LINKED ISSUE #968: VMEM on IMX6
When you use commands its breake on an assert in do_map_region() 
`assert(!mmu_pte_present(pte + pte_idx)); `

### PR DESCRIPTION
**Pull request by Deryugin**
Working on ARM MMU to fix #968 
Along the way, a number of vmem-related issues were fixed.

### CODE REVIEW

File: src/arch/arm/cortexa8/mmu.c [hunk 1]
```diff
@@ -48,18 +45,6 @@ static int mmu_init(void) {
 	);
 
 	return 0;
-
-	/* Setup physical address of the first level translation table */
-	__asm__ __volatile__ (
```
Comment (anton-bondarev) [inline src/arch/arm/cortexa8/mmu.c]: There isn't init for arm mmu?

Comment (Deryugin) [inline src/arch/arm/cortexa8/mmu.c]: There's setup of P15 register.
The physical address of the translation table is unique for each task and it's set on `mmu_set_context()`.

File: src/drivers/clock/integratorcp_clock.c [hunk 2]
```diff
@@ -45,6 +47,15 @@
 EMBOX_UNIT_INIT(integratorcp_init);
 
 static int integratorcp_clock_setup(struct time_dev_conf * conf) {
+	if (NULL == mmap_device_memory(
+		(void*) TIMER_BASE,
+		0x28,
```
Comment (anton-bondarev) [inline src/drivers/clock/integratorcp_clock.c]: Why region size initialization is different for integrator_clock & cortexa9_timer

File: third-party/lib/linenoise/linenoise.c [hunk 3]
```diff
@@ -662,8 +662,10 @@ static int linenoiseEdit(int fdin, int fdout, char *buf, size_t buflen, const ch
         switch(c) {
         case 10:    /* '\n' enter */
         case 13:    /* '\r' enter */
-            history_len--;
-            sysfree(history[history_len]);
+	    if (history_len) {
```
Comment (anton-bondarev) [inline third-party/lib/linenoise/linenoise.c]: This is right branch for these changes

File: src/drivers/clock/integratorcp_clock.c (hunk 2 above)
Comment (Deryugin) [inline src/drivers/clock/integratorcp_clock.c]: Thus size is just the highest register offset from periph base.

File: third-party/lib/linenoise/linenoise.c (hunk 3 above)
Comment (Deryugin) [inline third-party/lib/linenoise/linenoise.c]: In some templates there were no heap in config, so this variable was NULL. Without MMU it produced no errors, but with MMU on it led to exception

**Comment by anton-bondarev** [comment]
LGTM

**Pull request merged**